File: 刷题/golang/187. 重复的DNA序列.go
const L = 10

func findRepeatedDnaSequences(s string) (ans []string) {
    cnt := map[string]int{}
    for i := 0; i <= len(s)-L; i++ {
        sub := s[i : i+L]
        cnt[sub]++
        if cnt[sub] == 2 {
            ans = append(ans, sub)
        }
    }
    return
}

File: 刷题/cpp/187. 重复的DNA序列.cpp
class Solution {
public:
    vector<string> findRepeatedDnaSequences(string s) {
        unordered_map<string, int> mp;
        if(s.length()<=10){
            return {};
        }
        for(int i=0;i<=s.length()-10;i++){
            string str=s.substr(i,10);
            mp[str]++;
        }
        vector<string> ans;
        for(auto [k,v]: mp) {
            if(v>1){
                ans.push_back(k);
            }
        }
        return ans;
    }
};